ikarus/io/vtkwriter.hh

```diff
@@ -0,0 +1,182 @@
+
+  // Sets the VTK file format
+  void setFormat(Dune::Vtk::FormatTypes format) { writer_.setFormat(format); }
+
+  /// Sets the global datatype used for coordinates and other global float values
+  void setDatatype(Dune::Vtk::DataTypes datatype) { writer_.setDatatype(datatype); }
+
+  /// Sets the integer type used in binary data headers
+  void setHeadertype(Dune::Vtk::DataTypes datatype) { writer_.setHeadertype(datatype); }
+
+  auto& writer() const { return writer_; }
+  auto& writer() { return writer_; }
+
+  template <typename RF, Impl::Concepts::DataType DT>
+  void addResultFunction(RF&& resultFunction, DT /*type */) {
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<RF>(resultFunction));
+    else
+      writer_.addPointData(std::forward<RF>(resultFunction));
+  }
+
+  template <template <typename, int, int> class RT, Impl::Concepts::DataType DT>
+  requires(Concepts::ResultType<RT>)
+  void addResult(DT type) {
+    auto resFunction = makeResultVtkFunction<RT>(assembler_);
+    addResultFunction(std::move(resFunction), type);
+  }
+
+  template <Impl::Concepts::DataType DT>
+  void addAllResults(DT type) {
+    using ResultTuple = typename FEType::SupportedResultTypes;
+
+    Dune::Hybrid::forEach(ResultTuple(), [&]<typename RT>(RT i) { addResult<RT::template Rebind>(type); });
+  }
+
+  template <typename GF, Impl::Concepts::DataType DT>
+  void addGridFunction(GF&& gridFunction, const std::string& name, size_t size, DT /* type */) {
+    auto fieldInfo = Dune::Vtk::FieldInfo(name, size);
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<GF>(gridFunction), fieldInfo);
+    else
+      writer_.addPointData(std::forward<GF>(gridFunction), fieldInfo);
+  }
+
+  template <typename GF, typename FieldInfo, Impl::Concepts::DataType DT>
+  requires(std::same_as<FieldInfo, Dune::VTK::FieldInfo> or std::same_as<FieldInfo, Dune::Vtk::FieldInfo>)
+  void addGridFunction(GF&& gridFunction, FieldInfo fieldInfo, DT /* type */) {
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<GF>(gridFunction), fieldInfo);
+    else
+      writer_.addPointData(std::forward<GF>(gridFunction), fieldInfo);
+  }
+
+  template <int dim, typename R, typename Basis, Impl::Concepts::DataType DT>
+  void addInterpolation(R&& vals, const Basis& basis, const std::string& name, DT type) {
+    auto gridFunction = Dune::Functions::makeDiscreteGlobalBasisFunction<Dune::FieldVector<double, dim>>(basis, std::forward<R>(vals));
+    addGridFunction(std::move(gridFunction), name, dim, type);
+  }
+
+  // Other write functions are not explicitly exposed but can be access through writer(). ...
+  std::string write(const std::string& fn, std::optional<std::string> dir = {}) const { return writer_.write(fn, dir); }
+
+  // Legacy functions to add Cell and Point Data
+  template <class... Args>
+  void addCellData(Args&&... args) {
```

Comment:
If you would inherit publcily  from the underlying vtk writer these functions would be exposed outomatiucally.